src/main/java/dev/mccue/magicbean/processor/AnnotationProcessor.java

```diff
@@ -72,52 +72,88 @@ public boolean equals(Object o) {
                                         
                         """.formatted(
                         className,
-                        fields.stream()
-                                .map(field -> "java.util.Objects.equals(%s.%s, other.%s)".formatted(
-                                        selfExpr,
-                                        field.getSimpleName(),
-                                        field.getSimpleName()
-                                ))
-                                .collect(Collectors.joining(" && \n                   "))
+                        createEqualsExpression(selfExpr, fields)
                 )
         );
         equalsAndHashCodeMethods.append("""
                             @Override
                             public int hashCode() {
-                                return java.util.Objects.hash(
-                                  %s
-                                );
+                                %s
                             }
                         
                         """.formatted(
-                        fields.stream()
-                                .map(field -> "      " + selfExpr + "." + field.getSimpleName())
-                                .collect(Collectors.joining(",\n          "))
+                        createHashCodeMethodBody(selfExpr, fields)
                 )
         );
         return equalsAndHashCodeMethods.toString();
     }
 
+    private String createEqualsExpression(String selfExpr, List<VariableElement> fields) {
+        if (fields.isEmpty()) {
+            return "true";
+        }
+        else {
+            return fields.stream()
+                    .map(field -> "java.util.Objects.equals(%s.%s, other.%s)".formatted(
+                            selfExpr,
+                            field.getSimpleName(),
+                            field.getSimpleName()
+                    ))
+                    .collect(Collectors.joining(" && \n                   "));
+        }
+    }
+
+    private String createHashCodeMethodBody(String selfExpr, List<VariableElement> fields) {
+        if (fields.isEmpty()) {
+            return "return 1;";
+        }
+        else {
+            return """
+                    return java.util.Objects.hash(
+                                      %s
+                                    );
+                    """.formatted(
+                    fields.stream()
+                            .map(field -> "      " + selfExpr + "." + field.getSimpleName())
+                            .collect(Collectors.joining(",\n          ")));
+        }
```

Comment:
The indentation here for regular cases is a bit messed up now

![image](https://user-images.githubusercontent.com/5004262/151078377-53873fe9-346c-4772-a775-a1601b303e90.png)

Reply:
Fixed